src/top/dashboard.rs

```diff
@@ -200,22 +199,41 @@ impl<'a> Widgets<'a> {
             ];
 
             data.extend_from_slice(&formatted_metrics);
-            Row::new(data).style(Style::default())
-        });
+            items.push(Row::new(data).style(Style::default()));
+
+            // Add output rows. Note, we ignore outputs if it only contains
+            // [`DEFAULT_OUTPUT`] to avoid redundancy
+            if !(r.outputs.len() == 1 && r.outputs.contains_key(DEFAULT_OUTPUT)) {
+                for (id, output) in r.outputs.iter() {
+                    let sent_events_metric = format_metric(
+                        output.sent_events_total,
+                        output.sent_events_throughput_sec,
+                        self.opts.human_metrics,
+                    );
+                    let mut data = [""; NUM_COLUMNS]
+                        .into_iter()
+                        .map(Cell::from)
+                        .collect::<Vec<_>>();
+                    data[1] = Cell::from(id.as_ref());
+                    data[5] = Cell::from(sent_events_metric);
+                    items.push(Row::new(data).style(Style::default()));
+                }
+            }
+        }
 
         let w = Table::new(items)
             .header(Row::new(header).bottom_margin(1))
             .block(Block::default().borders(Borders::ALL).title("Components"))
             .column_spacing(2)
             .widths(&[
-                Constraint::Percentage(19),
-                Constraint::Percentage(14),
-                Constraint::Percentage(8),
-                Constraint::Percentage(8),
-                Constraint::Percentage(18),
-                Constraint::Percentage(18),
-                Constraint::Percentage(18),
-                Constraint::Percentage(8),
+                Constraint::Percentage(15), // ID
```

Comment:
Calling this bit out, these width constraints puzzle me a bit, so I'm happy to update. I couldn't find clear `tui-rs` documentation about them. 

Previously, there were `8` values which summed to `>100%` but only `7` columns (likely a small mistake?). When I changed them to account for the new `Output` column and sum to exactly `100%`, the `Errors` column started to get cut off entirely at certain screen widths. So, for now, they're set somewhat arbitrarily and sum to `<100`.

Reply:
I faced a version of this problem in an earlier implementation, and IIRC set a summed value < 100% for exactly that reason. It's somewhat arbitrary, based on trial and error.

The use of 8 values rather than 7 is possibly due to adding/removing columns along the way and overlooking this section. As long as data fits within a typical viewport, that's probably fine for now.

(I will also note that tui-rs has been updated since the first implementation, and may well have new abstractions to deal with width/sizing/column placement. I'd defer further exploration to a later, overall UI improvement phase.)